packages/theme/src/cli/utilities/theme-environment.ts

```diff
@@ -23,7 +23,7 @@ export async function startDevServer(theme: Theme, ctx: DevServerContext, onRead
 async function ensureThemeEnvironmentSetup(theme: Theme, ctx: DevServerContext) {
   if (ctx.themeEditorSync) {
     await initializeThemeEditorSync(theme, ctx.session, ctx.remoteChecksums, ctx.localThemeFileSystem)
-  } else {
-    await uploadTheme(theme, ctx.session, ctx.remoteChecksums, ctx.localThemeFileSystem, {})
   }
+
+  await uploadTheme(theme, ctx.session, ctx.remoteChecksums, ctx.localThemeFileSystem, {})
```

Comment:
Since we are handling theme upload here, I added a comment [here](https://github.com/Shopify/cli/pull/3894/files#diff-9eb3bb93a3f5e57984c31d76daeaa4528cd2fb87617a0fc28667502d3114a97fR11-R14) to clarify that the `remote-theme-watcher` does not upload files